procmonAnalyze.js

```diff
@@ -495,6 +479,39 @@ async function readFileContents() {
       return acc;
     }, {}));
 
+    data = data.map(row => {
+      let operation = row.operation;
+      let path = row.path;
+      let pid = row.pid;
+      let tid = row.tid;
+      let detail = row.detail;
+      let processName = row.processName;
+      let start = parseTimeString(row.time);
+      let duration = parseFloat(row.duration);
+      if (isNaN(duration)) {
+        duration = 0.01;
+      }
+      return {
+        operation, path, pid, tid, start, duration, detail, processName
+      };
+    }).filter(row => row.duration > 0 || row.operation == "Process Start");
+
+    let profilerData = null;
+    if (profilerInput.files[0]) {
+      reader.readAsText(profilerInput.files[0], "UTF-8");
+      let profilerText = await new Promise((resolve, reject) => {
+        reader.onload = e => {
+          resolve(e.target.result);
+        };
+        reader.onerror = e => {
+          reject("error reading file");
+        };
+      });
```

Comment:
Nit: could you add a commit extracting this chunk, and the two places above that do the same thing, into a helper function?